src/NumericalAlgorithms/LinearSolver/ExplicitInverse.hpp

```diff
@@ -109,10 +109,13 @@ class ExplicitInverse : public LinearSolver<LinearSolverRegistrars> {
    * re-creating the `begin()` iterator. The `reset()` function must not
    * invalidate the `end()` iterator.
    */
-  template <typename LinearOperator, typename VarsType, typename SourceType>
-  Convergence::HasConverged solve(gsl::not_null<VarsType*> solution,
-                                  const LinearOperator& linear_operator,
-                                  const SourceType& source) const noexcept;
+  template <typename LinearOperator, typename VarsType, typename SourceType,
+            typename... OperatorArgs>
+  Convergence::HasConverged solve(
+      gsl::not_null<VarsType*> solution, LinearOperator&& linear_operator,
```

Comment:
Why are you taking this as a universal reference?  You don't forward it anywhere.

Reply:
This way the argument preserves `const`-ness of the value that's passed in. If the `linear_operator` has a `const` call-operator, then it may be passed in as `const`. But if the `linear_operator` has a non-`const` call-operator, then it must be passed in as non-`const`. If there's a better way to handle this, I'm happy for suggestions!

Note that the subdomain operator currently has a non-const call-operator because it keeps memory buffers as member variables. I could make those memory buffers `mutable` to make the call-operator `const`. Not sure what's preferable.

Reply:
`gsl::not_null<LinearOperator*>`

C++'s type system is weird and unintuitive, but occasionally the rules make things simpler than they look like they should be.  Here, `LinearOperator` can be inferred as `const Foo` if necessary.  (This also works with mutable lvalue references, so you can `auto& x = db::get(...)` and get confusing errors when you try to use it.)

Reply:
Does that work if the `linear_operator` is a lambda function?

Reply:
Lambdas yes, but temporaries no, which probably is an issue.  OK, leave as-is.

Reply:
I can do that and make a bunch of private memory buffers in the subdomain operator `mutable`. Yes, @wthrowe?

Reply:
This is why `mutable` members exist, and I won't block such a change, but keep in mind that it means you have to be more careful using the object in multithreaded code.  We usually assume that const methods are thread-safe, but `mutable` members change that.

Of course, (smart) pointers can also break thread safety in const methods, and those cases are harder to detect.

Reply:
Ok I guess I'll leave this as-is then, unless you ask you me make a change.

Reply:
Yeah, that's fine.  Other fixups are fine, too.